six/modules/c++/six.sicd/include/six/sicd/Utilities.h

```diff
@@ -61,10 +61,26 @@ class Utilities
      * given GriddedDisplayType.
      * This always uses a PlanarGridECEFTransform
      * \param complexData ComplexData from which to construct fitter
+     * \param numPoints1D Number of points to use in each direction of grid.
      * \return ProjectionPolynomialFitter from ComplexData
      */
     static std::auto_ptr<scene::ProjectionPolynomialFitter>
-    getPolynomialFitter(const ComplexData& complexData);
+    getPolynomialFitter(const ComplexData& complexData,
+                        size_t numPoints1D =
+                         scene::ProjectionPolynomialFitter::DEFAULTS_POINTS_1D);
+
+    /*!
+     * Build ProjectionPolynomialFitter from complexData. Use the valid data
+     * polygon to constrain the grid points.
+     * \param complexData ComplexData from which to construct fitter
+     * \param numPoints1D Number of points to use in each direction of grid.
+     * \return ProjectionPolynomialFitter from ComplexData
+     */
+    static std::auto_ptr<scene::ProjectionPolynomialFitter>
```

Comment:
Rather than a separate function, add a bool to the existing function for whether or not to only sample within the VDP.  Default to the current behavior.